sparse/src/KokkosSparse_SortCrs.hpp

```diff
@@ -338,71 +56,144 @@ void sort_crs_matrix(const execution_space& exec, const rowmap_t& rowmap, const
   static_assert(!std::is_const_v<typename entries_t::value_type>,
                 "sort_crs_matrix: entries_t must not be const-valued");
   static_assert(!std::is_const_v<typename values_t::value_type>, "sort_crs_matrix: value_t must not be const-valued");
-  using lno_t    = typename entries_t::non_const_value_type;
-  using team_pol = Kokkos::TeamPolicy<execution_space>;
-  bool useRadix  = !KokkosKernels::Impl::kk_is_gpu_exec_space<execution_space>();
-  lno_t numRows  = rowmap.extent(0) ? rowmap.extent(0) - 1 : 0;
-  if (numRows == 0) return;
-  Impl::SortCrsMatrixFunctor<execution_space, rowmap_t, entries_t, values_t> funct(useRadix, rowmap, entries, values);
-  if (useRadix) {
-    Kokkos::parallel_for("sort_crs_matrix", Kokkos::RangePolicy<execution_space>(exec, 0, numRows), funct);
+  using Offset  = typename rowmap_t::non_const_value_type;
+  using Ordinal = typename entries_t::non_const_value_type;
+  using Scalar  = typename values_t::non_const_value_type;
+  // This early return condition covers having 0 or 1 entries,
+  // which is also implied by having 0 rows or 0 columns.
+  // If only 1 entry, the matrix is already sorted.
+  if (entries.extent(0) <= size_t(1)) {
+    return;
+  }
+  Ordinal numRows = rowmap.extent(0) ? rowmap.extent(0) - 1 : 0;
+  if constexpr (!KokkosKernels::Impl::kk_is_gpu_exec_space<execution_space>()) {
+    using UnsignedOrdinal   = typename std::make_unsigned<Ordinal>::type;
+    using entries_managed_t = Kokkos::View<typename entries_t::data_type, typename entries_t::device_type>;
+    using values_managed_t  = Kokkos::View<typename values_t::data_type, typename values_t::device_type>;
+    entries_managed_t entriesAux(Kokkos::view_alloc(Kokkos::WithoutInitializing, "Entries aux"), entries.extent(0));
+    values_managed_t valuesAux(Kokkos::view_alloc(Kokkos::WithoutInitializing, "Values aux"), values.extent(0));
+    // On CPUs, use a sequential radix sort within each row.
+    Kokkos::parallel_for(
+        "sort_crs_matrix", Kokkos::RangePolicy<execution_space, Kokkos::Schedule<Kokkos::Dynamic>>(exec, 0, numRows),
```

Comment:
Can we make the label more specific to know which code branch we are taking here